lib/queue.rb

```diff
@@ -1,31 +1,60 @@
 class Queue
 
   def initialize
-    # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+    @store = Array.new(21)
+    @front = -1
+    @back = -1
+    @size = 0
   end
 
   def enqueue(element)
-    raise NotImplementedError, "Not yet implemented"
+    return nil if((@front == 0 && @back == @store.length - 1) || (@back == (@front - 1 ) % (@store.size - 1))) 
+    @size += 1      
+    if (@front == -1) 
+      @front, @back = 0, 0
+    elsif(@back == (@store.length - 1) && @front != 0) 
```

Comment:
I don't think you need to check `@front != 0` here because you did that on line 11.